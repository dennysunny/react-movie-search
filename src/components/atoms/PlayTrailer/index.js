import React, { Component } from 'react';
import ModalVideo from 'react-modal-video';
import { propTypes } from './propTypes';

import 'react-modal-video/css/modal-video.min.css';

class PlayTrailer extends Component {
  static propTypes = propTypes;

  state = {
    isOpen: false,
  }

  openModal = () => {
    this.setState({ isOpen: true });
  }

  render() {
    const { isOpen } = this.state;
    const { video: { key } } = this.props;
    return (
      <div className="play-trailer-wrapper">
        <ModalVideo
          channel="youtube"
          isOpen={isOpen}
          videoId={key}
          onClose={() => this.setState({ isOpen: false })}
        />
        <button type="button" className="play-trailer" onClick={this.openModal}>
          <span className="fa fa-angle-right"> Watch Trailer </span>
        </button>
      </div>
    );
  }
}

export default PlayTrailer;
