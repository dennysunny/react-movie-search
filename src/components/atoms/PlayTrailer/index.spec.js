import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PlayTrailer from '.';

Enzyme.configure({ adapter: new Adapter() });

describe('Atoms/PlayTrailer', () => {
  it('should render as expected', () => {
    const video = {
      key: 'somekey',
    };
    const render = shallow(<PlayTrailer video={video} />);

    expect(render).toMatchSnapshot();
  });
});
