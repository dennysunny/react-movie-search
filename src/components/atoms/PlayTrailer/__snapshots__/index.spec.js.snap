// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/PlayTrailer should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayTrailer
    video={
      Object {
        "key": "somekey",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ModalVideo
          allowFullScreen={true}
          animationSpeed={300}
          aria={
            Object {
              "dismissBtnMessage": "Close the modal by clicking here",
              "openMessage": "You just openned the modal video",
            }
          }
          channel="youtube"
          classNames={
            Object {
              "modalVideo": "modal-video",
              "modalVideoBody": "modal-video-body",
              "modalVideoClose": "modal-video-close",
              "modalVideoCloseBtn": "modal-video-close-btn",
              "modalVideoEffect": "modal-video-effect",
              "modalVideoIframeWrap": "modal-video-movie-wrap",
              "modalVideoInner": "modal-video-inner",
            }
          }
          isOpen={false}
          onClose={[Function]}
          ratio="16:9"
          videoId="somekey"
          vimeo={
            Object {
              "api": false,
              "autopause": true,
              "autoplay": true,
              "byline": true,
              "callback": null,
              "color": null,
              "height": null,
              "loop": false,
              "maxheight": null,
              "maxwidth": null,
              "player_id": null,
              "portrait": true,
              "title": true,
              "width": null,
              "xhtml": false,
            }
          }
          youku={
            Object {
              "autoplay": 1,
              "show_related": 0,
            }
          }
          youtube={
            Object {
              "autoplay": 1,
              "cc_load_policy": 1,
              "color": null,
              "controls": 1,
              "disablekb": 0,
              "enablejsapi": 0,
              "end": null,
              "fs": 1,
              "h1": null,
              "iv_load_policy": 1,
              "list": null,
              "listType": null,
              "loop": 0,
              "modestbranding": null,
              "origin": null,
              "playlist": null,
              "playsinline": null,
              "rel": 0,
              "showinfo": 1,
              "start": 0,
              "theme": "dark",
              "wmode": "transparent",
            }
          }
        />,
        <button
          className="play-trailer"
          onClick={[Function]}
        >
          <span
            className="fa fa-angle-right"
          />
           Watch Trailer
        </button>,
      ],
      "className": "play-trailer-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFullScreen": true,
          "animationSpeed": 300,
          "aria": Object {
            "dismissBtnMessage": "Close the modal by clicking here",
            "openMessage": "You just openned the modal video",
          },
          "channel": "youtube",
          "classNames": Object {
            "modalVideo": "modal-video",
            "modalVideoBody": "modal-video-body",
            "modalVideoClose": "modal-video-close",
            "modalVideoCloseBtn": "modal-video-close-btn",
            "modalVideoEffect": "modal-video-effect",
            "modalVideoIframeWrap": "modal-video-movie-wrap",
            "modalVideoInner": "modal-video-inner",
          },
          "isOpen": false,
          "onClose": [Function],
          "ratio": "16:9",
          "videoId": "somekey",
          "vimeo": Object {
            "api": false,
            "autopause": true,
            "autoplay": true,
            "byline": true,
            "callback": null,
            "color": null,
            "height": null,
            "loop": false,
            "maxheight": null,
            "maxwidth": null,
            "player_id": null,
            "portrait": true,
            "title": true,
            "width": null,
            "xhtml": false,
          },
          "youku": Object {
            "autoplay": 1,
            "show_related": 0,
          },
          "youtube": Object {
            "autoplay": 1,
            "cc_load_policy": 1,
            "color": null,
            "controls": 1,
            "disablekb": 0,
            "enablejsapi": 0,
            "end": null,
            "fs": 1,
            "h1": null,
            "iv_load_policy": 1,
            "list": null,
            "listType": null,
            "loop": 0,
            "modestbranding": null,
            "origin": null,
            "playlist": null,
            "playsinline": null,
            "rel": 0,
            "showinfo": 1,
            "start": 0,
            "theme": "dark",
            "wmode": "transparent",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="fa fa-angle-right"
            />,
            " Watch Trailer",
          ],
          "className": "play-trailer",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-angle-right",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          " Watch Trailer",
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ModalVideo
            allowFullScreen={true}
            animationSpeed={300}
            aria={
              Object {
                "dismissBtnMessage": "Close the modal by clicking here",
                "openMessage": "You just openned the modal video",
              }
            }
            channel="youtube"
            classNames={
              Object {
                "modalVideo": "modal-video",
                "modalVideoBody": "modal-video-body",
                "modalVideoClose": "modal-video-close",
                "modalVideoCloseBtn": "modal-video-close-btn",
                "modalVideoEffect": "modal-video-effect",
                "modalVideoIframeWrap": "modal-video-movie-wrap",
                "modalVideoInner": "modal-video-inner",
              }
            }
            isOpen={false}
            onClose={[Function]}
            ratio="16:9"
            videoId="somekey"
            vimeo={
              Object {
                "api": false,
                "autopause": true,
                "autoplay": true,
                "byline": true,
                "callback": null,
                "color": null,
                "height": null,
                "loop": false,
                "maxheight": null,
                "maxwidth": null,
                "player_id": null,
                "portrait": true,
                "title": true,
                "width": null,
                "xhtml": false,
              }
            }
            youku={
              Object {
                "autoplay": 1,
                "show_related": 0,
              }
            }
            youtube={
              Object {
                "autoplay": 1,
                "cc_load_policy": 1,
                "color": null,
                "controls": 1,
                "disablekb": 0,
                "enablejsapi": 0,
                "end": null,
                "fs": 1,
                "h1": null,
                "iv_load_policy": 1,
                "list": null,
                "listType": null,
                "loop": 0,
                "modestbranding": null,
                "origin": null,
                "playlist": null,
                "playsinline": null,
                "rel": 0,
                "showinfo": 1,
                "start": 0,
                "theme": "dark",
                "wmode": "transparent",
              }
            }
          />,
          <button
            className="play-trailer"
            onClick={[Function]}
          >
            <span
              className="fa fa-angle-right"
            />
             Watch Trailer
          </button>,
        ],
        "className": "play-trailer-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFullScreen": true,
            "animationSpeed": 300,
            "aria": Object {
              "dismissBtnMessage": "Close the modal by clicking here",
              "openMessage": "You just openned the modal video",
            },
            "channel": "youtube",
            "classNames": Object {
              "modalVideo": "modal-video",
              "modalVideoBody": "modal-video-body",
              "modalVideoClose": "modal-video-close",
              "modalVideoCloseBtn": "modal-video-close-btn",
              "modalVideoEffect": "modal-video-effect",
              "modalVideoIframeWrap": "modal-video-movie-wrap",
              "modalVideoInner": "modal-video-inner",
            },
            "isOpen": false,
            "onClose": [Function],
            "ratio": "16:9",
            "videoId": "somekey",
            "vimeo": Object {
              "api": false,
              "autopause": true,
              "autoplay": true,
              "byline": true,
              "callback": null,
              "color": null,
              "height": null,
              "loop": false,
              "maxheight": null,
              "maxwidth": null,
              "player_id": null,
              "portrait": true,
              "title": true,
              "width": null,
              "xhtml": false,
            },
            "youku": Object {
              "autoplay": 1,
              "show_related": 0,
            },
            "youtube": Object {
              "autoplay": 1,
              "cc_load_policy": 1,
              "color": null,
              "controls": 1,
              "disablekb": 0,
              "enablejsapi": 0,
              "end": null,
              "fs": 1,
              "h1": null,
              "iv_load_policy": 1,
              "list": null,
              "listType": null,
              "loop": 0,
              "modestbranding": null,
              "origin": null,
              "playlist": null,
              "playsinline": null,
              "rel": 0,
              "showinfo": 1,
              "start": 0,
              "theme": "dark",
              "wmode": "transparent",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="fa fa-angle-right"
              />,
              " Watch Trailer",
            ],
            "className": "play-trailer",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-angle-right",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            " Watch Trailer",
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Atoms/PlayTrailer should render given video 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayTrailer
    video={
      Object {
        "key": "somekey",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ModalVideo
          allowFullScreen={true}
          animationSpeed={300}
          aria={
            Object {
              "dismissBtnMessage": "Close the modal by clicking here",
              "openMessage": "You just openned the modal video",
            }
          }
          channel="youtube"
          classNames={
            Object {
              "modalVideo": "modal-video",
              "modalVideoBody": "modal-video-body",
              "modalVideoClose": "modal-video-close",
              "modalVideoCloseBtn": "modal-video-close-btn",
              "modalVideoEffect": "modal-video-effect",
              "modalVideoIframeWrap": "modal-video-movie-wrap",
              "modalVideoInner": "modal-video-inner",
            }
          }
          isOpen={false}
          onClose={[Function]}
          ratio="16:9"
          videoId="somekey"
          vimeo={
            Object {
              "api": false,
              "autopause": true,
              "autoplay": true,
              "byline": true,
              "callback": null,
              "color": null,
              "height": null,
              "loop": false,
              "maxheight": null,
              "maxwidth": null,
              "player_id": null,
              "portrait": true,
              "title": true,
              "width": null,
              "xhtml": false,
            }
          }
          youku={
            Object {
              "autoplay": 1,
              "show_related": 0,
            }
          }
          youtube={
            Object {
              "autoplay": 1,
              "cc_load_policy": 1,
              "color": null,
              "controls": 1,
              "disablekb": 0,
              "enablejsapi": 0,
              "end": null,
              "fs": 1,
              "h1": null,
              "iv_load_policy": 1,
              "list": null,
              "listType": null,
              "loop": 0,
              "modestbranding": null,
              "origin": null,
              "playlist": null,
              "playsinline": null,
              "rel": 0,
              "showinfo": 1,
              "start": 0,
              "theme": "dark",
              "wmode": "transparent",
            }
          }
        />,
        <button
          className="play-trailer"
          onClick={[Function]}
        >
          <span
            className="fa fa-angle-right"
          />
           Watch Trailer
        </button>,
      ],
      "className": "play-trailer-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFullScreen": true,
          "animationSpeed": 300,
          "aria": Object {
            "dismissBtnMessage": "Close the modal by clicking here",
            "openMessage": "You just openned the modal video",
          },
          "channel": "youtube",
          "classNames": Object {
            "modalVideo": "modal-video",
            "modalVideoBody": "modal-video-body",
            "modalVideoClose": "modal-video-close",
            "modalVideoCloseBtn": "modal-video-close-btn",
            "modalVideoEffect": "modal-video-effect",
            "modalVideoIframeWrap": "modal-video-movie-wrap",
            "modalVideoInner": "modal-video-inner",
          },
          "isOpen": false,
          "onClose": [Function],
          "ratio": "16:9",
          "videoId": "somekey",
          "vimeo": Object {
            "api": false,
            "autopause": true,
            "autoplay": true,
            "byline": true,
            "callback": null,
            "color": null,
            "height": null,
            "loop": false,
            "maxheight": null,
            "maxwidth": null,
            "player_id": null,
            "portrait": true,
            "title": true,
            "width": null,
            "xhtml": false,
          },
          "youku": Object {
            "autoplay": 1,
            "show_related": 0,
          },
          "youtube": Object {
            "autoplay": 1,
            "cc_load_policy": 1,
            "color": null,
            "controls": 1,
            "disablekb": 0,
            "enablejsapi": 0,
            "end": null,
            "fs": 1,
            "h1": null,
            "iv_load_policy": 1,
            "list": null,
            "listType": null,
            "loop": 0,
            "modestbranding": null,
            "origin": null,
            "playlist": null,
            "playsinline": null,
            "rel": 0,
            "showinfo": 1,
            "start": 0,
            "theme": "dark",
            "wmode": "transparent",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="fa fa-angle-right"
            />,
            " Watch Trailer",
          ],
          "className": "play-trailer",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-angle-right",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          " Watch Trailer",
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ModalVideo
            allowFullScreen={true}
            animationSpeed={300}
            aria={
              Object {
                "dismissBtnMessage": "Close the modal by clicking here",
                "openMessage": "You just openned the modal video",
              }
            }
            channel="youtube"
            classNames={
              Object {
                "modalVideo": "modal-video",
                "modalVideoBody": "modal-video-body",
                "modalVideoClose": "modal-video-close",
                "modalVideoCloseBtn": "modal-video-close-btn",
                "modalVideoEffect": "modal-video-effect",
                "modalVideoIframeWrap": "modal-video-movie-wrap",
                "modalVideoInner": "modal-video-inner",
              }
            }
            isOpen={false}
            onClose={[Function]}
            ratio="16:9"
            videoId="somekey"
            vimeo={
              Object {
                "api": false,
                "autopause": true,
                "autoplay": true,
                "byline": true,
                "callback": null,
                "color": null,
                "height": null,
                "loop": false,
                "maxheight": null,
                "maxwidth": null,
                "player_id": null,
                "portrait": true,
                "title": true,
                "width": null,
                "xhtml": false,
              }
            }
            youku={
              Object {
                "autoplay": 1,
                "show_related": 0,
              }
            }
            youtube={
              Object {
                "autoplay": 1,
                "cc_load_policy": 1,
                "color": null,
                "controls": 1,
                "disablekb": 0,
                "enablejsapi": 0,
                "end": null,
                "fs": 1,
                "h1": null,
                "iv_load_policy": 1,
                "list": null,
                "listType": null,
                "loop": 0,
                "modestbranding": null,
                "origin": null,
                "playlist": null,
                "playsinline": null,
                "rel": 0,
                "showinfo": 1,
                "start": 0,
                "theme": "dark",
                "wmode": "transparent",
              }
            }
          />,
          <button
            className="play-trailer"
            onClick={[Function]}
          >
            <span
              className="fa fa-angle-right"
            />
             Watch Trailer
          </button>,
        ],
        "className": "play-trailer-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFullScreen": true,
            "animationSpeed": 300,
            "aria": Object {
              "dismissBtnMessage": "Close the modal by clicking here",
              "openMessage": "You just openned the modal video",
            },
            "channel": "youtube",
            "classNames": Object {
              "modalVideo": "modal-video",
              "modalVideoBody": "modal-video-body",
              "modalVideoClose": "modal-video-close",
              "modalVideoCloseBtn": "modal-video-close-btn",
              "modalVideoEffect": "modal-video-effect",
              "modalVideoIframeWrap": "modal-video-movie-wrap",
              "modalVideoInner": "modal-video-inner",
            },
            "isOpen": false,
            "onClose": [Function],
            "ratio": "16:9",
            "videoId": "somekey",
            "vimeo": Object {
              "api": false,
              "autopause": true,
              "autoplay": true,
              "byline": true,
              "callback": null,
              "color": null,
              "height": null,
              "loop": false,
              "maxheight": null,
              "maxwidth": null,
              "player_id": null,
              "portrait": true,
              "title": true,
              "width": null,
              "xhtml": false,
            },
            "youku": Object {
              "autoplay": 1,
              "show_related": 0,
            },
            "youtube": Object {
              "autoplay": 1,
              "cc_load_policy": 1,
              "color": null,
              "controls": 1,
              "disablekb": 0,
              "enablejsapi": 0,
              "end": null,
              "fs": 1,
              "h1": null,
              "iv_load_policy": 1,
              "list": null,
              "listType": null,
              "loop": 0,
              "modestbranding": null,
              "origin": null,
              "playlist": null,
              "playsinline": null,
              "rel": 0,
              "showinfo": 1,
              "start": 0,
              "theme": "dark",
              "wmode": "transparent",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="fa fa-angle-right"
              />,
              " Watch Trailer",
            ],
            "className": "play-trailer",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-angle-right",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            " Watch Trailer",
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
